objectherkenning_openbare_ruimte/on_edge/detection_pipeline/components/input_image.py

```diff
@@ -0,0 +1,64 @@
+import logging
+from typing import Dict, List
+
+import cv2
+import numpy as np
+
+logger = logging.getLogger("image")
+
+
+class InputImage:
+    def __init__(self, image_full_path: str):
+        self.image = cv2.imread(str(image_full_path))
+        self.mapx = None
+        self.mapy = None
+
+    def resize(self, output_image_size: List):
+        """Resizes the image
+
+        Parameters
+        ----------
+        output_image_size : List
+            Output size on format: [width, height]
+        """
+        self.image = cv2.resize(self.image, output_image_size)
+
+    def defisheye(self, defisheye_params: Dict):
+        """Removes fisheye effect from the images.
+
+        Parameters
+        ----------
+        defisheye_params : Dict
+            Parameters to use for defisheying. The structure is the following:
+                camera_matrix: [[2028, 0, 1954.1], [0, 2029.6, 1055.1], [ 0, 0, 1]]
+                distortion_params: [[-0.24083, 0.10647, 0.00083113, 0.0001802, -0.025874]]
+                input_image_size: [3840, 2160]
+        """
+        if self.mapx is None or self.mapy is None:
```

Comment:
Since this logic is now moved to the InputImage class, and a new instance of that class is created for each single image, we lose the efficiency gain from pre-computing these variables. In fact, pre-computing makes no sense anymore since you shouldn't defisheye the same image multiple times.

Reply:
Perhaps we move this logic back to the detection class, and only pass mapx and mapy here?

Or at least we should check the computational overhead of doing this for each image. Check one of the last log files, and compare the execution time of the first call to defisheye with any later call.

Reply:
Check now and let me know what do you think.